src/platform/plugins/shared/dashboard/public/dashboard_renderer/viewport/dashboard_viewport.tsx

```diff
@@ -124,15 +134,18 @@ export const DashboardViewport = ({
           <ExitFullScreenButton onExit={onExit} toggleChrome={!dashboardApi.isEmbeddedExternally} />
         </EuiPortal>
       )}
-      {panelCount === 0 && <DashboardEmptyScreen />}
       <div
         className={classes}
         data-shared-items-container
         data-title={dashboardTitle}
         data-description={description}
-        data-shared-items-count={panelCount}
+        data-shared-items-count={visiblePanelCount}
       >
-        <DashboardGrid dashboardContainerRef={dashboardContainerRef} />
+        {panelCount === 0 && sectionCount === 0 ? (
+          <DashboardEmptyScreen />
+        ) : (
+          <DashboardGrid dashboardContainerRef={dashboardContainerRef} />
+        )}
```

Comment:
This fixes a bug where the empty Dashboard screen would "slide in" after deleting the last panel on a Dashboard (due to the height smoother in `kbn-grid-layout`):

| Before | After |
|--------|--------|
| ![Apr-03-2025 09-03-13](https://github.com/user-attachments/assets/210d2397-f6bf-4a75-ace8-9e294f006fc9) | ![Apr-03-2025 09-10-36](https://github.com/user-attachments/assets/68aea28e-e1bc-4d65-ade2-eb9044ca23cd) |